common/cuda_hip/components/atomic.hpp.inc

```diff
@@ -147,6 +147,17 @@ GKO_BIND_ATOMIC_ADD(unsigned int);
 GKO_BIND_ATOMIC_ADD(unsigned long long int);
 GKO_BIND_ATOMIC_ADD(float);
 
+#if defined(__HIPCC__) && defined(__HIP_DEVICE_COMPILE__) && \
+    GINKGO_HIP_PLATFORM_HCC
+
+
+// the double atomicAdd is added after 4.3
+GKO_BIND_ATOMIC_ADD(double);
+
+
+#endif  // defined(__HIPCC__) && defined(__HIP_DEVICE_COMPILE__) &&
+        // GINKGO_HIP_PLATFORM_HCC
+
```

Comment:
Can we merge this with the CUDA condition, so we have only a single place where we control access to double atomics?

Reply:
yes in practice with considering __HIP_DEVICE_COMPILE__ is always true in atomic_add.hpp.inc when compiled by HIPCC.
otherwise, it will not reduce the condition.